framework/src/main/java/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -279,32 +276,63 @@ private static void initializeExtendsBound(
      */
     private static class InitializerVisitor implements AnnotatedTypeVisitor<Void, Void> {
         /**
-         * The BoundStructure starting from the first wildcard or type variable bound initialization
-         * that kicked this visitation off.
+         * The RecursiveTypeStructure corresponding to the first wildcard or type variable bound
+         * initialization that kicked this visitation off.
          */
-        private final BoundStructure topLevelStructure;
+        private final RecursiveTypeStructure topLevelStructure;
 
-        private BoundStructure currentStructure = null;
+        /**
+         * The {@link RecursiveTypeStructure} corresponding to the wildcard or type variable that is
+         * currently being visited.
+         */
+        private RecursiveTypeStructure currentStructure;
 
+        /** A mapping from TypeVariable to its {@link TypeVariableStructure}. */
         private final Map<TypeVariable, TypeVariableStructure> typeVarToStructure = new HashMap<>();
-        // private final Map<TypeVariable, TypeVariableRecord> typeVarToRecord = new HashMap<>();
+
+        /**
+         * A mapping from WildcardType to its {@link AnnotatedWildcardType}. The first time this
+         * visitor encounters a wildcard it creates an annotated type and adds it to this map. The
+         * next time the wilcard is encounter, the annotated type in this map is returned.
+         */
         private final Map<WildcardType, AnnotatedWildcardType> wildcards = new HashMap<>();
+
+        /**
+         * A mapping from IntersectionType to its {@link AnnotatedIntersectionType}. The first time
+         * this visitor encounters an intersection it creates an annotated type and adds it to this
+         * map. The next time the intersection is encounter, the annotated type in this map is
+         * returned.
+         */
         private final Map<IntersectionType, AnnotatedIntersectionType> intersections =
                 new HashMap<>();
+        /**
+         * Mapping from {@link TypeVariable} to {@link AnnotatedTypeMirror}. The annotated type
```

Comment:
Could you add a comment why this is a mapping to `AnnotatedTypeMirror` instead of `AnnotatedTypeVariable`?
From the previous two maps I expected an `AnnotatedTypeVariable` here.